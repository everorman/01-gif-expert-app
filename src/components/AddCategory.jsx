import { useState } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setCategories(inputValue);
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit}>
      <input type='text' placeholder='Buscar gif' value={inputValue} onChange={onInputChange} />
    </form>
  );
};

AddCategory.propsType = {
  setCategories: PropTypes.object.isRequired,
};
