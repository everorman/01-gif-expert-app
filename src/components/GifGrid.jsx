import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { getGifs } from '../helpers/getGifts';
export const GifGrid = ({ category }) => {
  useEffect(() => {
    getGifs(category);
  });
  return (
    <>
      <h3>{category}</h3>
      <h2>Hola mundo</h2>
    </>
  );
};

GifGrid.propsType = {
  category: PropTypes.string.isRequired,
};
