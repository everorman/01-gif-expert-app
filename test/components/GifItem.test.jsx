import { GifItem } from '../../src/components/GifItem';
import { render, screen, fireEvent, getByText } from '@testing-library/react';
describe('GifItem', () => {
  const title = 'Demo title';
  const url = 'http://demo.url.com/img.jpg';
  it('should do match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  it('should render img with url ant alt', () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  it('should render title', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
