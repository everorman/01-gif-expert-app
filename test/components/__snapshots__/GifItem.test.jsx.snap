// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Demo title"
      src="http://demo.url.com/img.jpg"
    />
    <p>
      Demo title
    </p>
  </div>
</div>
`;
